src/sentry/static/sentry/app/views/issueList/tagFilter.tsx

```diff
@@ -1,32 +1,44 @@
 import debounce from 'lodash/debounce';
-import PropTypes from 'prop-types';
 import React from 'react';
 
 import {Client} from 'app/api';
 import {addErrorMessage} from 'app/actionCreators/indicator';
 import {t, tct} from 'app/locale';
 import SelectControl from 'app/components/forms/selectControl';
+import {Tag, TagValue} from 'app/types';
 
-class IssueListTagFilter extends React.Component {
-  static tagValueToSelectFormat = ({value}) => ({
+import {TagValueLoader} from './types';
+
+const defaultProps = {
+  value: '',
+};
+
+type SelectOption = Record<'value' | 'label', string>;
+
+type Props = {
+  tag: Tag;
+  onSelect: (tag: Tag, value: string | null) => void;
+  tagValueLoader: TagValueLoader;
+} & typeof defaultProps;
+
+type State = {
+  query: string;
+  isLoading: boolean;
+  value: string | null;
+  textValue: string;
+  options?: SelectOption[];
+};
+
+class IssueListTagFilter extends React.Component<Props, State> {
+  static tagValueToSelectFormat = ({value}: TagValue): SelectOption => ({
```

Comment:
does this really have to be static on the class, can we just pull this function out?

Reply:
good call, it seems like it's only used once for a `map` anyways so I'll just make it an anonymous function inside of there.